### PR TITLE
Use expect_correction on more specs

### PR DESCRIPTION
**Pull request by bquorning**
---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [ ] Added the new cop to `config/default.yml`.
* [ ] The cop documents examples of good and bad code.
* [ ] The tests assert both that bad code is reported and that good code is not reported.

### CODE REVIEW

File: spec/rubocop/cop/rspec/hooks_before_examples_spec.rb [hunk 1]
```diff
@@ -33,6 +50,14 @@
         ^^^^^^^^^^^^^^^^ Move `before` above the examples in the group.
       end
     RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        before { setup }
+        include_examples('should be after let')
```
Comment (Darhazer) [inline spec/rubocop/cop/rspec/hooks_before_examples_spec.rb]: guess we should reword this example to use `before` and not `let`. Now I see how you find that those two cops are nearly identical :)

File: spec/rubocop/cop/rspec/multiple_subjects_spec.rb [hunk 2]
```diff
@@ -19,6 +19,20 @@
         end
       end
     RUBY
+
+    expect_correction [
```
Comment (Darhazer) [inline spec/rubocop/cop/rspec/multiple_subjects_spec.rb]: any reason a heredoc couldn't be used here?

Comment (bquorning) [reply spec/rubocop/cop/rspec/multiple_subjects_spec.rb]: Yeah, the code currently in line 27 (`      '  ',`) would contain trailing spaces if formatted as a heredoc.

File: spec/rubocop/cop/rspec/hooks_before_examples_spec.rb (hunk 1 above)
Comment (bquorning) [reply spec/rubocop/cop/rspec/hooks_before_examples_spec.rb]: 😆 

Fixed

**Comment by bquorning** [comment]
Another review @Darhazer ? (please merge on +1)

File: spec/rubocop/cop/rspec/multiple_subjects_spec.rb (hunk 2 above)
Comment (Darhazer) [reply spec/rubocop/cop/rspec/multiple_subjects_spec.rb]: Guess we can improve the autocorrect by using `range_with_surrounding_space` but that can happen in another PR.
I wonder how it was working before with the `include_examples` though

Comment (bquorning) [reply spec/rubocop/cop/rspec/multiple_subjects_spec.rb]: > I wonder how it was working before with the `include_examples` though

In the previous examples there are two `include_examples('autocorrect', …`. In the first example, two `subject`s are changed to `let`, and heredoc is used. In the second example, two unnamed `subject`s are *removed*, and the indentation is left behind as trailing spaces – and heredoc isn’t used in that example.

And yes, we might be able to improve it in another PR. But I would assume people will have cops guarding against trailing spaces, empty lines etc.

**Pull request merged**